third_party/proton/dialect/lib/ProtonGPUToLLVM/PatternProtonGPUOpToLLVM.cpp

```diff
@@ -10,28 +9,32 @@
 namespace mlir::triton {
 namespace proton::gpu {
 
-namespace {
-struct CircularStoreOpConversion
-    : public ConvertOpToLLVMPattern<
-          mlir::triton::proton::gpu::CircularStoreOp> {
-  explicit CircularStoreOpConversion(
-      LLVMTypeConverter &typeConverter,
-      const proton::gpu::TargetInfoBase &targetInfo, PatternBenefit benefit)
-      : mlir::ConvertOpToLLVMPattern<
-            mlir::triton::proton::gpu::CircularStoreOp>(typeConverter, benefit),
-        targetInfo(targetInfo) {}
+Value getLinearId(Location loc, ConversionPatternRewriter &rewriter,
+                  ModuleOp mod, const proton::gpu::TargetInfoBase &targetInfo) {
+  auto &tritonTargetInfo = targetInfo.getTritonTargetInfo();
+  auto b = TritonLLVMOpBuilder(loc, rewriter);
+  // Note: we compute use i64 data type to compute and then truncate to i32 to
+  // support various backend intrinsics (e.g. amd).
+  Value pidX =
```

Comment:
Cannot we just use the GPU dialect? e.g., `gpu.block_id`

Reply:
OK, I get it now. It's actually because Triton's target info had this `programId` interface before without using the GPU dialect. Let's keep it for now in Proton. I'll check if I can make replace it with `gpu.block_id` instead.